InternxtDesktop/Views/UIKit/AppButton.swift

```diff
@@ -104,24 +120,58 @@ struct AppButton: View {
     public let onClick: () -> Void
     public var type: AppButtonType = .primary
     public var size: AppButtonSize = .MD
-    
+    public var icon: AppIconName?
+    @Binding public var isEnabled: Bool
+
+    init(title: String, onClick: @escaping () -> Void, type: AppButtonType = .primary, size: AppButtonSize = .MD, isEnabled: Binding<Bool> = .constant(true)) {
+        self.title = title
+        self.onClick = onClick
+        self.type = type
+        self.size = size
+        self._isEnabled = isEnabled
+    }
+
+    init(icon: AppIconName, title: String, onClick: @escaping () -> Void, type: AppButtonType = .primary, size: AppButtonSize = .MD, isEnabled: Binding<Bool> = .constant(true)) {
+        self.title = title
+        self.icon = icon
+        self.onClick = onClick
+        self.type = type
+        self.size = size
+        self._isEnabled = isEnabled
```

Comment:
what's this property? isn't this the same as `isEnabled`?

Reply:
to initialize a Binding variable in SwiftUI you have to assign that initial value to its storage first. that's why i wrote it that way

https://sarunw.com/posts/binding-initialization/

Reply:
interesting, thanks!